diff --git a/configure.py b/configure.py
index a9910d038f0..bc8f548b807 100755
--- a/configure.py
+++ b/configure.py
@@ -1673,7 +1673,7 @@ def configure_node(o):
   o['variables']['want_separate_host_toolset'] = int(cross_compiling)
 
   if target_arch == 'arm64':
-    o['variables']['arm_fpu'] = options.arm_fpu or 'neon'
+    pass
 
   if options.node_snapshot_main is not None:
     if options.without_node_snapshot:
