diff --git a/src/builtins/builtins-typed-array.cc b/src/builtins/builtins-typed-array.cc
index b2cd7e53315..6abd034342a 100644
--- a/src/builtins/builtins-typed-array.cc
+++ b/src/builtins/builtins-typed-array.cc
@@ -819,15 +819,9 @@ BUILTIN(Uint8ArrayPrototypeToBase64) {
     // 11. Return CodePointsToString(outAscii).
 
     size_t simd_result_size;
-    if (uint8array->buffer()->is_shared()) {
-      simd_result_size = simdutf::atomic_binary_to_base64(
-          std::bit_cast<const char*>(uint8array->GetBuffer()->backing_store()),
-          length, reinterpret_cast<char*>(output->GetChars(no_gc)), alphabet);
-    } else {
       simd_result_size = simdutf::binary_to_base64(
           std::bit_cast<const char*>(uint8array->GetBuffer()->backing_store()),
           length, reinterpret_cast<char*>(output->GetChars(no_gc)), alphabet);
-    }
     DCHECK_EQ(simd_result_size, output_length);
     USE(simd_result_size);
   }
